src/components/UI/ControllerHints/index.tsx

```diff
@@ -0,0 +1,163 @@
+        main = t('controller.hints.play_game', 'Play game')
+      }
+    } else if (target.id === 'search') {
+      // focusing the search bar
+      main = t(
+        'controller.hints.open_virtual_keyboard',
+        'Open virtual keyboard'
+      )
+    } else if (target.closest('.MuiMenu-list')) {
+      // focusing a context menu on a card
+      main = t('controller.hints.activate', 'Activate')
+      alt = t('controller.hints.close_context_menu', 'Close context menu')
+    } else if (classes.contains('hg-button')) {
+      // focusing a virtual keyboard element
+      main = t('controller.hints.activate', 'Activate')
+      back = t('controller.hints.close_keyboard', 'Close keyboard')
+      alt = t('controller.hints.space', 'Space')
+      alt2 = t('controller.hints.backspace', 'Backspace')
+    }
+
+    setMainActionHint(main)
+    setAltActionHint(alt)
+    setAltActionHint2(alt2)
+    setBackActionHint(back)
+  }
+
+  // proxy method to set hints depending on what's focused
+  const setHints = (target: HTMLElement | null | typeof globalThis) => {
+    // sometimes focus can be null or the window object
+    if (!target || target === globalThis) {
+      setMainActionHint('')
+      setAltActionHint('')
+      setAltActionHint2('')
+      setBackActionHint('')
+      return
+    }
+
+    setHintsFor(target as HTMLElement)
+  }
+
+  // listen to `controller-changed` custom events to set the controller brand
+  // listen to `focus` events to detect the current focused element
+  useEffect(() => {
+    const onControllerChanged = (e: CustomEvent<{ controllerId: string }>) => {
+      setController(e.detail.controllerId)
+    }
+    window.addEventListener('controller-changed', onControllerChanged)
+
+    const onFocusChanged = (e: FocusEvent) => {
+      const tgt = !e.target ? null : (e.target as HTMLElement)
+      setHints(tgt)
+    }
+
+    window.addEventListener('focus', onFocusChanged, true)
+
+    return () => {
+      window.removeEventListener('controller-changed', onControllerChanged)
+      window.removeEventListener('focus', onFocusChanged)
+    }
+  }, [])
+
+  useEffect(() => {
+    // check if there's any page to go back
+    if (history.state.idx === 0) {
+      setBackActionFallback('')
+    } else {
+      setBackActionFallback(t('controller.hints.back', 'Back'))
+    }
+
+    // check focused element after a page change
+    setHints(document.querySelector(':focus') as HTMLElement)
+  }, [location])
+
+  useEffect(() => {
+    // set the brand for the images to use
+    if (controller.match(/sony|0ce6|PS3|PLAYSTATION|0268|'2563.*0523/i)) {
+      setLayout('ps')
+    } else if (controller.match(/28de.*11ff/)) {
+      setLayout('steam-deck')
+    } else if (controller.match(/microsoft|xbox/i)) {
+      setLayout('xbox')
+    } else {
+      setLayout('steam-deck')
+    }
+  }, [controller])
+
+  // empty if no controller activated
+  if (controller === '') {
```

Comment:
```suggestion
  if (!controller) {
```

Small stuff but that would do it as well and it's a bit cleaner imo.